apollo-gradle-plugin/src/main/java/com/apollographql/android/gradle/ApolloClassGenTask.java

```diff
@@ -37,7 +40,11 @@ void generateClasses(IncrementalTaskInputs inputs) {
     inputs.outOfDate(new Action<InputFileDetails>() {
       @Override
       public void execute(InputFileDetails inputFileDetails) {
-        new GraphQLCompiler().write(inputFileDetails.getFile(), outputDir, customTypeMapping, hasGuavaDep);
+        NullableValueType nullableValueType = NullableValueType.NONE;
```

Comment:
I think it would be better if we keep this logic part of the compiler.

Might as well create a data class `GraphQLCompilerArgs` to wrap all the arguments and doing these checks in `GraphQLCompiler.write()`?